onnxruntime/core/providers/shared_library/provider_host_api.h

```diff
@@ -1,5 +1,7 @@
 // Copyright (c) Microsoft Corporation. All rights reserved.
 // Licensed under the MIT License.
+//
+#include "core/framework/provider_options.h"
```

Comment:
Can you forward declare it here?

Reply:
Is it forward declare here?
In provider_options.h, I think they are all complete type:

      using ProviderOptions = std::unordered_map<std::string, std::string>;
      using ProviderOptionsVector = std::vector;
      using ProviderOptionsMap = std::unordered_map<std::string, ProviderOptions>;

Reply:
ProviderOptions can be forward declared in this header. Under onnxruntime you can say
struct/class ProviderOptions.